fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -257,6 +257,166 @@ sebasedBiasFieldMNI="$SubjectFolder/$AtlasSpaceFolder/Results/$NameOffMRI/${Name
+  log_Warn "  In HCP data 30% of the cortical surface will be misaligned by at least half cortical "
+  log_Warn "  thickness and 10% of the cortical surface will be completely misaligned by a full "
+  log_Warn "  cortical thickness. At a future time, we may be able to add support for fieldmap-less "
+  log_Warn "  distortion correction. At this time, however, despite ongoing efforts, this problem is"
+  log_Warn "  unsolved and no extant approach has been successfully shown to demonstrate clear "
+  log_Warn "  improvement according to the accuracy standards of HCP-Style data analysis when compared"
+  log_Warn "  to gold-standard fieldmap-based correction."
+fi
+
+# -- Slice timing correction
+
+if [ "${DoSliceTimeCorrection}" = 'TRUE' ]; then
+  ComplianceMsg+=" --doslicetime=TRUE --slicetimerparams=${SliceTimerCorrectionParameters}"
+  Compliance="LegacyStyleData"
+  log_Warn "This LegacyStyleData option of slice timing correction is performed before motion"
+  log_Warn "  correction (as is typically done in legacy-style brain imaging) and thus assumes that the"
+  log_Warn "  brain is motionless. Errors in temporal interpolation will occur in the presence of head"
+  log_Warn "  motion and may also disrupt data quality measures as shown in Power et al 2017 PLOS One "
+  log_Warn "  'Temporal interpolation alters motion in fMRI scans: Magnitudes and consequences for"
+  log_Warn "  artifact detection.' Slice timing correction and motion correction would ideally be performed"
+  log_Warn "  simultaneously; however, this is not currently supported by any major software tool. HCP-Style"
+  log_Warn "  fast TR fMRI data acquisitions (TR<=1s) avoid the need for slice timing correction, provide"
+  log_Warn "  major advantages for fMRI denoising, and are recommended."
+fi
+
+
+# -- Use of nonstandard BOLD mask
+
+if [ "${BOLDMask}" != 'T1_fMRI_FOV' ]; then
+  if [ "${BOLDMask}" != "T1_DILATED_fMRI_FOV" ] && [ "${BOLDMask}" != "T1_DILATED2x_fMRI_FOV" ] && [ "${BOLDMask}" != "fMRI_FOV" ] ; then
+    log_Err_Abort "--boldmask=${BOLDMask} is invalid! Valid options are: T1_fMRI_FOV (default), T1_DILATED_fMRI_FOV, T1_DILATED2x_fMRI_FOV, fMRI_FOV."
+  fi
+  ComplianceMsg+=" --boldmask=${BOLDMask}"
+  Compliance="LegacyStyleData"
+fi
+
+# -- Use of external BOLD reference
+
+if [ "$fMRIReference" = "NONE" ]; then
+  fMRIReferenceReg="NONE"    
+  fMRIReferencePath="NONE"
+  ReferenceResultsFolder="NONE"
+else
+  fMRIReferenceReg=`opts_DefaultOpt $fMRIReferenceReg "linear"`
+
+  # set reference and check if external reference (if one is specified) exists 
+
+  fMRIReferencePath="$Path"/"$Subject"/"$fMRIReference"
+  log_Msg "Using reference image from ${fMRIReferencePath}"
+  fMRIReferenceImage="$fMRIReferencePath"/"$ScoutName"_gdc
+  fMRIReferenceImageMask="$fMRIReferencePath"/"$ScoutName"_gdc_mask
+  ReferenceResultsFolder="$Path"/"$Subject"/"$AtlasSpaceFolder"/"$ResultsFolder"/"$fMRIReference"
+
+  if [ "$fMRIReferencePath" = "$fMRIFolder" ] ; then
```

Comment:
Given where we ended up, we should make clear in the comments above the `--fmriscout` and `--fmriref` are exclusionary options, and I think we should add an abort if both of those are set to something (other than NONE), since in that case the user is clearly confused about the distinction between the two options.

Reply:
Do you feel strongly about aborting or would it be enough to provide the following warning for the `--fmriref` in the code and then a similar logged warning if both `--fmriref` and `--fmriscout` are provided:

```
# WARNING: This option excludes tbe use of --fmriscout option, as the scout from the specified reference BOLD is used
```

Reply:
Yes, I feel fairly strongly that should be an abort condition.  While the warnings are well and good, the log files are very verbose, and I doubt that most users will ever notice the warnings in them.  If a user specifies both `--fmriref` and `--fmriscout` within the current code, then they are clearly confused as to what those options represent, and I feel that warrants an abort so that they can make sure they are running the code as they intended.

Reply:
Ok. I'll implement it that way then. In Qu|Nex the omission of the `--fmriscout` when generating the call will be handled automatically for the user, though, as what user wants is specified on a higher level. Qu|Nex will specify `--fmriscout` when calling the pipelines for the run used as the reference run and not for the others that then use it as a reference.